src/App.tsx

```diff
@@ -17,18 +94,40 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                text={text}
+                handleFilter={handleFilter}
+                handleClearInput={handleClearInput}
+                handleCheckStatus={handleCheckStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={visibleTodos}
+                  handleShowModal={handleShowModal}
+                  choseTodo={choseTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {showModal && (
```

Comment:
```suggestion
      {choseTodo && (
```
render only when selected todo exists